### PR TITLE
Make sure Angular CLI middleware supports "ng serve --ssl"

### PR DESCRIPTION
**Pull request by gunsh**
This fixes inability of middleware to listen on "ng serve --ssl" which solves #27790

Addresses #27790

### CODE REVIEW

File: src/Middleware/SpaServices.Extensions/src/AngularCli/AngularCliMiddleware.cs [hunk 1]
```diff
@@ -48,27 +48,19 @@ internal static class AngularCliMiddleware
...
-                return targetUriTask.WithTimeout(timeout,
+                return angularCliServerInfoTask.WithTimeout(timeout,
                     $"The Angular CLI process did not start listening for requests " +
                     $"within the timeout period of {timeout.TotalSeconds} seconds. " +
                     $"Check the log output for error information.");
             });
         }
 
-        private static async Task<AngularCliServerInfo> StartAngularCliServerAsync(
+        private static async Task<Uri> StartAngularCliServerAsync(
```
Comment (campersau) [inline src/Middleware/SpaServices.Extensions/src/AngularCli/AngularCliMiddleware.cs:63]: The `AngularCliServerInfo` class seems to be unused now and could be removed.

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/aspnetcore?pullRequest=27799) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/aspnetcore?pullRequest=27799) before we can accept your contribution.<br/><br/>:x: gunsh [sign now](https://cla.dotnetfoundation.org/dotnet/aspnetcore?pullRequest=27799)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/aspnetcore?pullRequest=27799) it.</sub>

File: src/Middleware/SpaServices.Extensions/src/AngularCli/AngularCliMiddleware.cs (hunk 1 above)
Comment (gunsh) [reply src/Middleware/SpaServices.Extensions/src/AngularCli/AngularCliMiddleware.cs:63]: I have just removed it.

Comment (asadsahi) [reply src/Middleware/SpaServices.Extensions/src/AngularCli/AngularCliMiddleware.cs:63]: This fix isn't working as reproduced with the code of this PR in this repo:

https://github.com/asadsahi/dotnet-angular11

**Comment by captainsafia** [comment]
Thanks for taking the time to put together this PR, @gunsh! 

It seems like this PR now requires that the Angular development server be started with SSL enabled?

Also, I notice this change (https://github.com/angular/angular-cli/pull/19412) has been merged into the Angular CLI. Does the issue still repro with this change?

**Comment by asadsahi** [comment]
@captainsafia possibly that may fix the issue, but will have to wait until next angular cli release. I wanted to raise this issue on angular repo first, but when I tried angular cli 11 project on its own it worked fine. 

Making angular cli started in ssl just for this error to go away is not ideal.

This issue only occurs when using angular 11 with .net 5. Therefore, my thinking is still that .net spa middleware has some fix in it.

**Comment by captainsafia** [comment]
Got it. This change seems functionally fine. Can you share the scenarios it was validated against? For example, does it work with versions of Angular other than Angular 11.

**Review by captainsafia — APPROVED** [review]

**Comment by captainsafia** [comment]
From the referenced issue, it looks like updating the CLI version resolves the issue with SSL connections so I don't think we should backport this to 5.0.

But this change (reading the URL of the app from the output) is worthwhile enough to include in 6.0, IMO.

**Pull request merged**